mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -375,6 +388,146 @@ pub(crate) async fn block_header_utxo_loop<T: UtxoCommonOps>(
     }
 }
 
+#[derive(Debug, Clone, Display, Eq, PartialEq)]
+pub enum ChainReorgError {
+    #[display(fmt = "Can't get from the storage for {} - reason: {}", coin, reason)]
+    BlockNotFound {
+        coin: String,
+        reason: String,
+    },
+    #[display(
+        fmt = "Unable to delete block headers from_height{from_height} to_height {to_height} from storage for \
+    {coin}\
+     - reason: \
+    {reason}"
+    )]
+    UnableToDeleteHeaders {
+        from_height: u64,
+        to_height: u64,
+        coin: String,
+        reason: String,
+    },
+    AddToStorageErr(String),
+}
+
+// `resolve_chain_reorg` is used to resolve chain reorganizations (chain splits) in a blockchain.
+// It takes in parameters including the Electrum client, the coin ticker being analyzed, a block header storage object,
+// the previous height of the blockchain, a pointer to the last block header, and a mutable hashmap of RPC headers.
+
+// The function first checks the hashmap of RPC headers for the latest header and sets it as the latest_rpc_header.
+// It then loops through the headers to find a conflicting header and checks if the bits (difficulty) of the conflicting headers match. If they do, the function removes the bad headers from rpc_headers and continues to use the headers from storage. If the bits don't match, the function checks the cumulative difficulty of the headers from RPC and the ones in storage to choose which chain is the best. If the headers from RPC have a higher difficulty, the function removes the bad headers from storage and continues to use the headers from RPC. If the headers from storage have a higher difficulty, the function keeps headers from storage and skips the conflicting header from RPC.
+
+// The function returns an error if it is unable to find a block header or delete headers from storage. If it
+// successfully resolves the chain reorganization, the function returns Ok(()).
+async fn resolve_chain_reorg(
+    client: &ElectrumClient,
+    storage: &BlockHeaderStorage,
+    previous_height: u64,
+    last_block_header: &ChainReorgHeader,
+    rpc_headers: &mut HashMap<u64, BlockHeader>,
+) -> Result<(), ChainReorgError> {
+    let ticker = last_block_header.coin.as_str();
+    let current_hash = last_block_header.hash;
+    let current_bits = u32::from(last_block_header.bits.clone());
+
+    let mut latest_rpc_header =
+        rpc_headers
+            .clone()
+            .into_values()
+            .next()
+            .ok_or_else(|| ChainReorgError::BlockNotFound {
+                coin: ticker.to_string(),
+                reason: "empty header".to_string(),
+            })?;
+    let mut next_height = previous_height;
+    let mut limit = 30;
+
+    while current_hash != latest_rpc_header.previous_header_hash || limit > 1 {
```

Comment:
Because of the above comments, this loop needs changing, I believe we should just remove the reorganized headers and they will be requested again in the next loop of `block_header_utxo_loop` if we continue instead of breaking as stated in one of the comments.

Reply:
alright, I can `continue` in the loop instead of break. also if a `header` rpc is the bad header I can clear all the downloaded `header` and re-fetch(restart the loop with `continue`)